File: 9. Intervals/56. Merge Intervals/Solution.cpp
#include <vector>

class Solution {
public:
    std::vector<std::vector<int>> merge(std::vector<std::vector<int>>& intervals) {
        std::sort(intervals.begin(), intervals.end());

        std::vector<std::vector<int>> res;

        for (const std::vector<int>& interval : intervals)
        {
            if (!res.empty() && interval[0] <= res.back()[1])
            {
                res.back()[1] = std::max(res.back()[1], interval[1]);
            }
            else
            {
                res.push_back(interval);
            }
        }

        return res;
    }
};

File: 9. Intervals/56. Merge Intervals/Solution.py
from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()

        res = [intervals[0]]

        for i in range(1, len(intervals)):
            a = res[-1]
            b = intervals[i]

            if b[0] <= a[1]:
                res[-1] = [a[0], max(a[1], b[1])]
            else:
                res.append(b)
        
        return res
    